pkgs/tools/typesetting/tex/texlive/build-texlive-package.nix

```diff
@@ -0,0 +1,144 @@
+{ lib
+, fetchurl
+, runCommand
+
+  # script interpteers
+, bash
+, jdk
+, perl
+, python3
+, ruby
+, snobol4
+, tk
+
+  # compiled binaries
+, texliveBinaries
+  # fixed hashes
+, texliveFixedHashes
+  # mirrors
+, texliveMirrors
+}:
+
+pname:
+
+{ revision
+, version ? toString revision
```

Comment:
Why is this not a string already?

Reply:
It happens to be an integer in `tlpdb.nix` due to how it's generated. Possibly to shave some kB's? I don't know if there is a particular reason.

Reply:
Saving kilobytes to lose them afterwards in a conversion? Strange.

Reply:
I mean, I think they were trying to save as much as possible on the storage size of `tlpdb.nix` inside Nixpkgs. I remember there was even a comment about using a truncated sha1 as fixed hash, but it must have been ages ago. I suppose we don't have this problem now. Still, I'd rather not touch `tlpdb.nix` right now to keep the PR small. I am happy to do it later, especially if it improves the evaluation speed.

(For clarity, the code inside `buildTeXPackage` is essentially the same code that was in `default.nix`, with only minor simplifications.)

Reply:
This doesn't really affect the eval performance that much. When instantiating `texlive.combined.scheme-full` with NIX_SHOW_STATS=1, the counters are slightly lower for the string version but cpuTime really is a tossup.
```patch
--- rev-int	2023-08-07 19:04:15.787714194 +0200
+++ rev-string	2023-08-07 19:03:36.381784589 +0200
@@ -1,6 +1,6 @@
 warning: you did not specify '--add-root'; the result might be removed by the garbage collector
 {
-  "cpuTime": 1.6834319829940796,
+  "cpuTime": 1.6877000331878662,
   "envs": {
     "bytes": 102700800,
     "elements": 5518694,
@@ -8,20 +8,20 @@
   },
   "gc": {
     "heapSize": 402976768,
-    "totalBytes": 556993968
+    "totalBytes": 556585280
   },
   "list": {
     "bytes": 12032672,
     "concats": 213548,
     "elements": 1504084
   },
-  "nrAvoided": 4542424,
+  "nrAvoided": 4520078,
   "nrFunctionCalls": 3402597,
   "nrLookups": 2688855,
   "nrOpUpdateValuesCopied": 8227231,
   "nrOpUpdates": 509776,
-  "nrPrimOpCalls": 1786462,
-  "nrThunks": 5364736,
+  "nrPrimOpCalls": 1763192,
+  "nrThunks": 5363812,
   "sets": {
     "bytes": 184097232,
     "elements": 10708881,
@@ -38,7 +38,7 @@
     "number": 50927
   },
   "values": {
-    "bytes": 144933528,
-    "number": 6038897
+    "bytes": 144911352,
+    "number": 6037973
   }
 }
 ```